TODO

```diff
@@ -0,0 +1,29 @@
+- Set permissions on the named pipe/Unix socket correctly. No need to let
+  random users spy on worktrees' activity to which they have no read access.
+
+- Skip reporting directories (meaning: make sure we know the type of a path)
+  - To do that, be smart and look whether a path refers to a file in a
+    subdirectory, then make sure that the subdirectory is marked as such
+  - Maybe over-report, and only mark known directories?
+  - Or not worry about this for now, it's okay to report directories, use
+    Trace2 to measure the impact of this
+
+- Exclude submodules somehow.
+
+- Handle when a file is removed and a directory added in its place: this needs
+  to be reported as "deleted".
```

Comment:
I'm not quite sure how we deal with this. @jeffhostetler do you know off the top of your head?

Reply:
@dscho  I'm not sure what you meant in your original TODO item.  Currently we will see the FS delete event for the path and report that path (without a trailing slash) back to the client (without any indication of why or what happened), so the client will have to look it up in the cache-entries.  We should also get an event for the directory creation which the daemon should report to the client (with a trailing slash) and that should trigger stuff in the untracked-cache.  So I think we're good here.  I'll add a task to create unit tests to confirm this.